### PR TITLE
Ft gif crud api

### PR DESCRIPTION
**Pull request by Ebunwadi**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
 import * as dotenv from 'dotenv';
 import express from 'express';
 import cors from 'cors';
+import fileupload from 'express-fileupload';
 import userApi from './routes/userApi.js';
+import gifApi from './routes/gifApi.js';
```
Comment (courage173) [inline server.js]: rename this to userRoutes and gitRoutes

File: controllers/gifApiLogic.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    if (!title) return res.status(400).json({ message: 'gif caption is required' });
+    const result = await cloudinary.v2.uploader.upload(image.tempFilePath, {
+      folder: 'gifs',
+    });
+    const { secure_url: secureUrl, public_id: publicId, created_at: createdOn } = result;
+    const postedBy = req.user.email;
+    const gifDB = await pool.query(
+      `INSERT INTO gifs (title, image_url, created_on, public_id, posted_by) 
+          VALUES ($1, $2, $3, $4, $5) RETURNING *`,
+      [title, secureUrl, createdOn, publicId, postedBy],
```
Comment (courage173) [inline controllers/gifApiLogic.js]: postedBy should simply be named userId

File: database/db.sql [hunk 3]
```diff
@@ -13,3 +13,15 @@ CREATE TABLE users (
     is_admin BOOLEAN
 );
 
+CREATE TABLE gifs (
+    id SERIAL PRIMARY KEY,
+    title VARCHAR(255) NOT NULL,
+    image_url VARCHAR(255) NOT NULL,
+    public_id VARCHAR(255) NOT NULL,
+    created_on timestamp with time zone NOT NULL,
+    posted_by VARCHAR(50) NOT NULL,
```
Comment (courage173) [inline database/db.sql]: rename posted_by to user_id

**Review by courage173 — CHANGES_REQUESTED** [review]
Looks good! left small comments. please ignore the temp folder on gitignore

File: server.js (hunk 1 above)
Comment (Ebunwadi) [reply server.js]: done

File: controllers/gifApiLogic.js (hunk 2 above)
Comment (Ebunwadi) [reply controllers/gifApiLogic.js]: done

File: database/db.sql (hunk 3 above)
Comment (Ebunwadi) [reply database/db.sql]: done

**Pull request merged**